File: src/main/python/Graphs/easy.py
from collections import deque

class Easy:

    @staticmethod
    def bfs_traversal(n:int, E:list[list[int]], initial_node:int) -> list:
        bfs_list = []

        if initial_node >= n:
            return bfs_list

        bfs_queue = deque()
        visited = [False]*n

        bfs_queue.append(initial_node)
        visited[initial_node] = True
        while bfs_queue:
            curr_node = bfs_queue.popleft()
            for node in E[curr_node]:
                if not visited[node]:
                    visited[node] = True
                    bfs_queue.append(node)
            bfs_list.append(curr_node)

        return bfs_list

    @staticmethod
    def dfs_traversal(n:int, E:list[list[int]], initial_node:int) -> list:
        dfs_list = []

        if initial_node >= n:
            return dfs_list

        stack = []
        visited = [False]*n

        stack.append(initial_node)
        visited[initial_node] = True
        while stack:
            curr_node = stack.pop()
            for node in E[curr_node]:
                if not visited[node]:
                    stack.append(node)
                    visited[node] = True
            dfs_list.append(curr_node)

        return dfs_list

    @staticmethod
    def count_connected_components() -> int:
        pass

File: src/main/java/Graphs/Easy.java
package main.java.Graphs;

import java.util.List;
import java.util.ArrayList;
import java.util.Queue;
import java.util.LinkedList;
import java.util.Deque;
import java.util.ArrayDeque;

public class Easy {
    public static List<Integer> bfsTraversal(int n, int[][] E, int initialNode) {
        List<Integer> bfsList = new ArrayList<>();

        if (initialNode >= n)
            return bfsList;

        Queue<Integer> traversalQueue = new LinkedList<>();
        boolean[] visited = new boolean[n];

        traversalQueue.offer(initialNode);
        visited[initialNode] = true;
        while (!traversalQueue.isEmpty()) {
            int currNode = traversalQueue.poll();
            for (int node : E[currNode]) {
                if (!visited[node]) {
                    traversalQueue.offer(node);
                    visited[node] = true;
                } 
            } bfsList.add(currNode);
        }

        return bfsList;
    }

    public static List<Integer> dfsTraversal(int n, int[][] E, int initialNode) {
        List<Integer> dfsList = new ArrayList<>();

        if (initialNode >= n)
            return dfsList;

        Deque<Integer> dfsStack = new ArrayDeque<>();
        boolean[] visited = new boolean[n];

        dfsStack.push(initialNode);
        visited[initialNode] = true;

        while (!dfsStack.isEmpty()) {
            int currNode = dfsStack.pop();
            dfsList.add(currNode);

            for (int node : E[currNode]) {
                if (!visited[node]) {
                    visited[node] = true;
                    dfsStack.push(node);
                }
            }
        }

        return dfsList;
    }

    public static int countConnectedComponents(int n, int[][] E) {
        return 1;
    }
}
